bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/EdCompTargetList.java

```diff
@@ -293,33 +362,40 @@ private void refreshAll() {
         final PositionType[] ptA = new PositionType[2 + guiders.size()];
         ptA[index++] = BasePositionType.instance;
         for (GuideProbe guider : guidersList) {
-            ptA[index++] = new GuidePositionType(guider, guider != illegal);
+            final boolean guiderNotIllegal = !illegal.exists(guider::equals);
+            ptA[index++] = new GuidePositionType(guider, guiderNotIllegal);
         }
         ptA[index] = UserPositionType.instance;
 
         _w.tag.removeActionListener(_tagListener);
         _w.tag.setModel(new DefaultComboBoxModel<>(ptA));
-        _w.tag.setEnabled(isEnabled() && (env.getBase() != _curPos));
+        _w.tag.setEnabled(isEnabled() && (!selectionIsBasePosition()));
         _w.tag.addActionListener(_tagListener);
 
         _w.tag.setRenderer(tagRenderer);
         showTargetTag();
 
         // Update target details
-        _w.detailEditor.edit(getObsContext(env), _curPos, getNode());
+        _curSelection.foreach(either -> either.forEachLeft(t -> _w.detailEditor.edit(getObsContext(env), t, getNode())));
```

Comment:
I see this pattern in several places.  We might want to add `toOption` to `ImEither`.  It would give you a `Some<R>` if right and a `None` otherwise.  Then we might define methods like:

````java
public Option<SPTarget> selectedTarget() {
   return _curSelection.swap().toOption();
}

public Option<IndexedGuideGroup> selectedGroup() {
    return _curSelection.toOption();
}
````

Then this can be simplified to

````java
  selectedTarget().foreach(t -> _w.detailEditor(getObsContext(env), t, getNode()));
````

I think it might be easier to read that way.